File: ThaiBubbles-H6/Model/ProductList.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ThaiBubbles_H6.Model
{
    public class ProductList
    {
        [Key]
        public int ProductOrderListID { get; set; }

        public double Quantity { get; set; } = 0;

        public int? OrderId { get; set; }
        public Order? Orders { get; set; }

        public int? ProductId { get; set; }
        public Product? Products { get; set; }
    }
}

File: ThaiBubbles-H6-Client/src/app/Models/ProductList.ts
import { Product } from './Product';
import { Order } from './Order';

export interface ProductList {
  productOrderListID: number;
  quantity: number;
  productId: number;
  Product?: Product; // Should be a single object, not an array
  orderId: number;
  Order?: Order; // Should be a single object, not an array
}
